lib/msf/core/auxiliary/redis.rb

```diff
@@ -10,8 +10,30 @@ module Auxiliary::Redis
     include Auxiliary::Scanner
     include Auxiliary::Report
 
-    REDIS_UNAUTHORIZED_RESPONSE = /(?<auth_response>ERR operation not permitted|NOAUTH Authentication required)/i
+    # This modules contains the responses that are send back by the Redis server
+    # depending on which command is being executed, and whether the current user
+    # is allowed to do so, or is required to be authenticated.
+    #
+    # The module is divided into 2 sub-modules with the same constants defined.
+    # Depending on which version of Redis is being tested, the correct sub-module
+    # constant should be used for comparing the response.
+    module Responses
+      # This module contains the responses for Redis versions 5 and older.
+      module RedisOld
+        AUTHENTICATION_REQUIRED = /(?<auth_response>NOAUTH Authentication required)/i
+        NO_PASSWORD_SET = /-ERR Client sent AUTH, but no password is set/i
+        WRONG_PASSWORD = /-ERR invalid password/i
+        OKAY = /\+OK/i
+      end
 
+      # This module contains the responses for Redis versions 6 and newer
+      module RedisNew
```

Comment:
This won't scale well if they Redis changes the response format again in $nextVersion; i.e. will this become RedisNewNew or such

Potentially we should add the version here

```ruby
class RedisV1
class RedisV6
```

(Assuming it was V1 that added the the 'Old' response format)

Reply:
Not a blocker, since this is existing code - but for visibility we have a `proto` namespace for implementing protocols. A chunk of this code would have been better suited to live there

Reply:
I couldn't find which version introduced the `requirepass` option, so pinning that to a version will be tricky.
The ACL Support is definitely introduced with the V6 of Redis, that one is written out in the documentation.

Where is the `proto` namespace located? Because I couldn't find anything directly for this?